src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -6285,7 +6285,7 @@ int RGWRados::get_obj_state_impl(const DoutPrefixProvider *dpp, RGWObjectCtx *oc
 
   if (r == -ENOENT) {
     s->exists = false;
-    s->has_attrs = true;
+    s->has_attrs = false;
```

Comment:
can you please clarify what breaks as a result of this?

looking back through git history, i see that it's been this way since 2011 though i'm not sure why: https://github.com/ceph/ceph/commit/b03930fa02295f8068195f30312ff2d6fac50991

Reply:
actually i have no idea why  [https://github.com/ceph/ceph/commit/b03930fa02295f8068195f30312ff2d6fac50991](url) set s->has_attrs = true either？